shared/mathutil/math_helper.go

```diff
@@ -90,3 +91,10 @@ func Min(a uint64, b uint64) uint64 {
 	}
 	return b
 }
+
+// MulOverflows returns true if the multiplication of the two values
+// is an overflow for uint64.
+func MulOverflows(a uint64, b uint64) bool {
+	h, _ := bits.Mul64(a, b)
```

Comment:
why not return `bool` and the result(64-bit) here. Otherwise you multiply twice using this method to only check.

Reply:
Or nvm ignore this, if we are being defensive here then mutiplying twice is not too big a deal.

Reply:
Agreed